src/chain/keysinterface.rs

```diff
@@ -235,4 +242,9 @@ impl KeysInterface for KeysManager {
 		sha.result(&mut seed);
 		ChannelKeys::new_from_seed(&seed)
 	}
+
+	fn get_session_key(&self) -> SecretKey {
+		let child_ix = self.session_child_index.fetch_add(1, Ordering::AcqRel);
+		self.session_master_key.ckd_priv(&self.secp_ctx, ChildNumber::from_hardened_idx(child_ix as u32)).expect("Your RNG is busted").secret_key
```

Comment:
Oops, sorry, this isnt sufficient as it'll wrap back to 0 on restart and we'll re-use session keys. Can you hash in the timestamp like we do for channel keys? I'm not too worried about needing extra rng output, here, though.